dash/include/dash/Matrix.h

```diff
@@ -417,30 +417,32 @@ class Matrix
   inline void                 barrier() const;
 
   /**
-   * Complete all outstanding non-blocking operations executed by all units
-   * on the array's underlying global memory.
+   * Complete all outstanding non-blocking operations to all units
+   * on the container's underlying global memory.
    *
    * \see  DashContainerConcept
    */
   inline void                 flush();
 
   /**
-   * Complete all outstanding non-blocking operations executed by the
-   * local unit on the narray's underlying global memory.
+   * Complete all outstanding non-blocking operations to the specified unit
```

Comment:
Really cool feature. This fixes / answers #416.

Reply:
Not entirely, unfortunately. We still need a beefed up `dash::Future` for bulk wait to not leak memory through the handles. It's on my list :)